menuinst/win32.py

```diff
@@ -115,6 +115,17 @@ def quoted(s):
         return s
 
 
+def ensure_pad(name, pad="_"):
+    """
+
+    Examples:
+        >>> ensure_pad('conda')
+        '_conda_'
+
+    """
+    return name and "%s%s%s" % (pad, name.strip(pad), pad)
```

Comment:
This will replace multiple instances of  pad  with a single instance, which will be an issue if the arguments are already quoted to multiple levels:

``` Python
>>> "_a_".strip("_")
'a'
>>> "__a__".strip("_")
'a'
```

Trying to figure out a nice solution to this.

Reply:
In the first commit on this PR I had

```python
if not name or name[0] == name[-1] == pad:
    return name
else:
    return "%s%s%s" % (pad, name, pad)
```

Reply:
How about the following:

``` Python
def ensure_pad(s, pad="_"):
    if s[0] != pad:
        s = pad + s
    if s[-1] != pad:
        s += pad
    return s
```

The only question is if created an unbalanced number of quotes will create an issue, which I think it will

Reply:
I would go with the original:

```
if not name or name[0] == name[-1] == pad:
    return name
else:
    return "%s%s%s" % (pad, name, pad)
```

That will never add unbalanced quotes and doesn't strip existing cases of multiple quotes

Reply:
Ok original added back.  And also re-opened https://github.com/conda/conda/pull/6817 because I want to keep these two functions consistent.